### ISSUE TITLE
Is Cfitsio needed?

### ISSUE THREAD

**Issue by jhmatthews** [sirocco-rt/sirocco#232]
Do we need Cfitsio in the code as it is? It takes quite a while to install and means there's additional room for error

As I can see no reason why it's used in "everyday" compilation of the code, I'm going to suggest we remove it from the standard install but allow it to be installed with an option in the Makefile, and so still ship it with the code. Thoughts? @kslong in particular?

**Comment by kslong** [comment]
I think we should answer the question posed, which is “Is Cfitsio needed” and not adopt some half way solution (an optional make), which is likely to cause problems in the future.

First, we should check whether there is any possibility of calling cfitsio.  I don’t remember.  I think it may have only been used in some of the helper programs, before we had astropy to construct images.

If it is called, or has the possibility of being called, then we should check whether it makes any sense today to keep that code.  If the answer is that we don’t think it is useful, then we should delete both cfitsio and the vestigial code.

If cfitisio is called and we think it has use, I would favor keeping cfitsio in the code tree, getting a more current version, and compiling it by default.  Cfitsio is fully maintaed by GSFC, and has never in my experience caused a problem in terms of compiling it, so I don’t think ease of compilation should be the determining factor.

**Comment by jhmatthews** [comment]
I agree Knox, I guess I just wanted to check a) why it's there in the first place and b) if there was a future need for it. For example, do we anticipate wanting to write data to fits file in the near future.

If not, I can't find any reference to "fitsio", "fitsio.h" or anything similar in the source code. Here are the header files included:

```
#include "atomic.h"
#include "python.h"
#include "recipes.h"
#include "models.h"
#include "templates.h"
#include "log.h"
#include "mpi.h"
#include <stdlib.h>
#include <stdio.h>
#include <math.h>
#include <strings.h>
#include <string.h>
#include <stdarg.h>
#include <gsl/gsl_sf_expint.h>
#include <gsl/gsl_block.h>
#include <gsl/gsl_vector.h>
#include <gsl/gsl_matrix.h>
#include "my_linalg.h"
#include <time.h>
#include <sys/time.h>
```

None of this appear to be cfitsio routines, so I think we can simply delete the cfitsio compilation step and relevant flags to link the library.

**Comment by kslong** [comment]
That seems pretty conclusive.  Let’s go ahead and pull cfitsio out altogether.

I am fairly sure that the reason it was in the routine originally is that I did not have good ways to make images of things like t_e, before python, and matplotlib showed up.  So I believe there were routines in py_wind or its subs that would create a fits image that could be displayed in ds9.

But that was then and this is now, so we should be able to let cfitsio go.

K

**Comment by jhmatthews** [comment]
sounds good. will sort this out tomorrow and remove. If we want to use it in the future we will only have to look back at older versions of git to understand how it used to be linked.

**Comment by jhmatthews** [comment]
closed by #236
